### PR TITLE
test: added test for mergeDeletionCallBack

### PR DESCRIPTION
**Pull request by satyazzz123**
## test for mergeDeletionCallBack
ii) test for scenario which checks for permissions of the destination directory to match the permissions of the source file when both the source file and destination directory exist

fixes a part of #881

### CODE REVIEW

**Comment by satyazzz123** [comment]
@kmehant @seshapad please review this PR.

File: filesystem/merger_test.go [hunk 1]
```diff
@@ -48,6 +48,50 @@ func TestMergeDeletionCallBack(t *testing.T) {
...
+		t.Fatalf("Failed to get destination directory info: %v", destErr)
+	}
+
+	// Check if the destination directory permissions match the source file
+	sourceFileInfo, sourceFileErr := os.Stat(sourceFile.Name())
+	if sourceFileErr != nil {
+		t.Fatalf("Failed to get source file info: %v", sourceFileErr)
+	}
+
+	if destInfo.Mode().Perm() != sourceFileInfo.Mode().Perm() {
```
Comment (HarikrishnanBalagopal) [inline filesystem/merger_test.go:91]: Also check if destination is indeed a directory

File: filesystem/merger_test.go [hunk 2]
```diff
@@ -48,6 +48,50 @@ func TestMergeDeletionCallBack(t *testing.T) {
...
+	defer os.Remove(sourceFile.Name())
+
+	// Create a temporary destination directory
+	destinationDir, err := ioutil.TempDir("", "destinationDir")
+	if err != nil {
+		t.Fatalf("Failed to create destination directory: %v", err)
+	}
+	defer os.RemoveAll(destinationDir)
+
+	// Call the merge function
```
Comment (HarikrishnanBalagopal) [inline filesystem/merger_test.go:73]:
```suggestion
	// Call the merge deletion callback function
```

**Review by HarikrishnanBalagopal — CHANGES_REQUESTED** [review]

**Comment by satyazzz123** [comment]
@HarikrishnanBalagopal I have addressed the requested changes . Please have a look . Thank you

**Review by HarikrishnanBalagopal — APPROVED** [review]
LGTM

**Pull request merged**